### ISSUE TITLE
Enhancement: Clear Notification for Subscriber Import Limit Exceeded

### ISSUE THREAD

**Issue by alinclamba** [Automattic/jetpack#38657]
### Impacted plugin

Jetpack

### What

Add a clear notification system added to the Jetpack Newsletter free plan to inform users when they have exceeded the 100-subscriber import limit

### How

**Current Behavior:**

- Free users can import up to 100 subscribers in total.
- When attempting to import or add more than 100 subscribers, users receive a message stating the list is being processed without any indication that the import will fail due to exceeding the limit.
- There is no notification that explains the 100-subscriber limit has been reached.

**Expected Behavior:**
Maybe we could implement a notification system that:
- Alerts users immediately when they attempt to import more than the allowed 100 subscribers.
- Provides a clear message explaining that the import limit for the free plan has been exceeded.- Optionally, offers suggestions for upgrading to a plan with a higher import limit.

**Suggestions:**
We could either show a message stating: "You have reached the maximum import limit of 100 subscribers for the free plan." or follow up with an email notification explaining the import limit and suggesting alternative actions (e.g., upgrading the plan), or both, or something else. 🙂

This has been discussed on Slack here: p1722418184740219-slack-C0CMN0V97

**Comment by chatoxz-wp** [comment]
Hey, @Automattic/loop, this was under the Zap team, but as they are not working on Newsletter anymore, I am moving it to you.

**Comment by robertbpugh** [comment]
Thanks for forwarding. We just discovered this same problem (and many others - pg9MHR-p8-p2) and will be working on improving this.

**Comment by allilevine** [comment]
@spsiddarthan shipped an import error notice in https://github.com/Automattic/wp-calypso/pull/99162 You'll now see this alert when you've exceeded the limit:

<img width="1136" alt="Image" src="https://github.com/user-attachments/assets/382734d7-8e53-4996-901d-c2e746897388" />

But it's still not obvious in the Substack importer flow.

**Comment by ollierozdarz** [comment]
Hi team,

Just confirming — are these the two locations where we should surface the Free plan subscriber limit notice (100 subscribers max)?

**1. Substack Importer**

Noting in this view, we're showing a callout to the 100 subscriber limit already, albeit in the section above the file upload dropzone.

![Image](https://github.com/user-attachments/assets/c1c85ddd-0c79-459f-91e4-5295d5e44e00)

![Image](https://github.com/user-attachments/assets/9c6e3d61-7c25-4bf3-87e6-659dbf3ba66a)

**2. Adding newsletter subscribers from CSV**

![Image](https://github.com/user-attachments/assets/0039b867-5604-4dd5-93f2-415df28e7f4d)

**Comment by ollierozdarz** [comment]
I've explored this and pushed some designs to [Figma](https://www.figma.com/design/kDH1Er46XVZyz3iKR7KSCE/Importer-Subscriber-Limit?node-id=0-1&t=YhfUQSmPYoMx2c6c-1). 

**TL;DR;**

I think it's best to show this notice in three places:

1. Add Subscribers Manually
2. Use a CSV File
3. Substack Importer Flow.

I'd opt to show the notice right above the Import dropzone (or form fields) since it's subtle, clear, and hits the user at exactly the right moment without cluttering the modal. Keeping the notice visually consistent in multiple locations aligns with the work happening to build sustainably with core and A8c components.

### Add Subscribers Manually

![Image](https://github.com/user-attachments/assets/650170a0-c4d9-4990-84ee-6067fcaf5b1f)

### Use a CSV File

![Image](https://github.com/user-attachments/assets/d78ee3dd-b3b8-4011-bd20-ebf1dc2f2e97)

### Substack Importer Flow

![Image](https://github.com/user-attachments/assets/512ec92b-45fb-4335-9aa3-514569836835)

**Note:** I've removed the current notice from the Instructional text, as [seen above](https://github.com/Automattic/jetpack/issues/38657#issuecomment-2764784712).

**Comment by robertbpugh** [comment]
This looks great to me.

**Comment by davemart-in** [comment]
Removing from Tier 2 bugs and moving to pe7F0s-2DF-p2 project thread.

**Comment by Addison-Stavlo** [comment]
@ollierozdarz - did we still want a support link in this notice somewhere? I see we replaced the link to support docs about the limit with a link to upgrade.<br><br>Il kick this off by also making the "up to 100 subscribers" part linked to the support doc as before, which opens the help center at:

<img src="https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/7a394ca9-a223-4a5d-b7fc-617a9d8c004f/dc25670d-bad7-4654-978d-5706ac0cf01f?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi83YTM5NGNhOS1hMjIzLTRhNWQtYjdmYy02MTdhOWQ4YzAwNGYvZGMyNTY3MGQtYmFkNy00NjU0LTk3OGQtNTcwNmFjMGNmMDFmIiwiaWF0IjoxNzQ5NTU0MTU5LCJleHAiOjMzMzIwMTE0MTU5fQ.x2jrVT6AiBIgxHxb1ZeLVIUBibTzEa_bmNn08Roecm0 " alt="Screenshot 2025-06-10 at 7.15.18 AM.png" width="456" data-linear-height="44" />

<img src="https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/79f8628b-279a-4ad2-af52-7b4750b34a43/b2ed1eb6-aff9-4c5a-b1de-362d4d662b31?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi83OWY4NjI4Yi0yNzlhLTRhZDItYWY1Mi03YjQ3NTBiMzRhNDMvYjJlZDFlYjYtYWZmOS00YzVhLWIxZGUtMzYyZDRkNjYyYjMxIiwiaWF0IjoxNzQ5NTU0MTU5LCJleHAiOjMzMzIwMTE0MTU5fQ.fjNG3ErBqxlmxPu5wsAYxLNto5CGRHx5wGt2jnam8io " alt="Screenshot 2025-06-10 at 7.09.27 AM.png" width="420" data-linear-height="301" />

But let me know if removal of that link was intentional and it seems unnecessary.

**Comment by Addison-Stavlo** [comment]
Another small change to the design I am currently working with if you think it makes sense, is the placement of the notice in the uploader modal.<br><br>It looks like we currently have conditional notices that appear above the upload Copy, such as the stale import notice:

<img src="https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/82a2d17d-cc2a-4540-adcd-a883f0fc1838/70551a18-cb43-4acb-bdf0-ca5420133f10?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi84MmEyZDE3ZC1jYzJhLTQ1NDAtYWRjZC1hODgzZjBmYzE4MzgvNzA1NTFhMTgtY2I0My00YWNiLWJkZjAtY2E1NDIwMTMzZjEwIiwiaWF0IjoxNzQ5NTU3NzkwLCJleHAiOjMzMzIwMTE3NzkwfQ.Dus0nQn9Clr5O0QkZvlV1yHMsdV0yRAoIc4ctmgiySk " alt="Screenshot 2025-06-10 at 8.07.01 AM.png" width="554" data-linear-height="270" />

Given that, would it make sense to move our free plan notice to the same place? (as opposed to beneath the longer copy)

<img src="https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/7cdfe956-ed4a-46d2-ba19-b938da3f731c/f36b6f4e-43b6-45ba-a560-6ba7a004cb08?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi83Y2RmZTk1Ni1lZDRhLTQ2ZDItYmExOS1iOTM4ZGEzZjczMWMvZjM2YjZmNGUtNDNiNi00NWJhLWE1NjAtNmJhN2EwMDRjYjA4IiwiaWF0IjoxNzQ5NTU3NzkwLCJleHAiOjMzMzIwMTE3NzkwfQ.Yuc2cFa9c3LYygj-_I8u_faVpWoRX0NyOjsqNgd72mo " alt="Screenshot 2025-06-10 at 8.05.48 AM.png" width="559" data-linear-height="587" />

It looks like the copy and everything below it come from an automattic package outside of calypso itself, which may be why notices added in calypso are placed above it. However, if we really want the notices below that copy I could alternatively make some adjustments to the package to allow us to pass some items through on the calypso side to be placed between that generic copy and the dropzone.

Im going with the above for now, but always happy to update further if needed.

**Comment by davemart-in** [comment]
That makes sense to me @Addison-Stavlo

**Comment by Addison-Stavlo** [comment]
https://github.com/Automattic/wp-calypso/pull/104101 should resolve.

**Comment by Addison-Stavlo** [comment]
and a follow up here - [https://github.com/Automattic/wp-calypso/pull/104214](https://github.com/Automattic/wp-calypso/pull/104214)
